packages/playwright-core/src/server/browserContext.ts

```diff
@@ -513,17 +514,88 @@ export abstract class BrowserContext extends SdkObject {
     };
     const originsToSave = new Set(this._origins);
 
+    async function _collectStorageScript() {
+
+      const idbResult = await Promise.all((await indexedDB.databases()).map(async dbInfo => {
+        if (!dbInfo.name)
+          throw new Error('Database name is empty');
```

Comment:
Let's make sure any error clearly states that it was not able to serialize IndexedDB. Probably a big try/catch around IndexedDB code would be the best.

Reply:
i've added `.catch` to the two big promises.